File: 프로그래머스/lv2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        
        // 오름차순 정렬 -> [0, 1, 3, 5, 6]
        Arrays.sort(citations);
        
        for (int i = 0; i < citations.length; i++) {
            if (citations[i] >= citations.length - i) {
                answer = citations.length - i;
                break;
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv2/42747. H－Index/H－Index.py
def solution(citations):     
    # 리스트 내림차순 정렬 [6, 5, 3, 1, 0]
    citations.sort(reverse=True)
    # [(0,6), (1,5), (2,3), (3,1), (4,0)]
    for idx, citation in enumerate(citations):
        if idx >= citation:
            return idx
    return len(citations)